.github/workflows/on_pull_request.yml

```diff
@@ -0,0 +1,41 @@
+name: On Pull Request
+
+on:
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - synchronize
+  merge_group:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint-and-test:
+    name: Lint and test
+    runs-on: ubuntu-latest
```

Comment:
We may need to change this in future to use a bigger runner

Reply:
I think we should do it now actually, so we don't forget this. ubuntu-latest-8core-x64 is the runner to use

Reply:
That's the agreed runner for Android Linux builds :)